### ISSUE TITLE
Question about Figure 3 in the paper

### ISSUE THREAD

**Issue by yuxineverforever** [lllyasviel/ControlNet#648]
Hi authors, I have a small question about the following piece of code and Figure 3.

```
1.        for module, zero_conv in zip(self.input_blocks, self.zero_convs):
2.            if guided_hint is not None:
3.                h = module(h, emb, context)
4.                h += guided_hint
5.                guided_hint = None
```

If I understand correctly, `h` should be the input z_t in Figure 3, `guided_hint` should be condition `c_f` in Figure 3.

Why we feed `h` through `module` and then add with `guided_hint` (Line 3-4)? According to Figure 3, shouldn't we add `h` with `guided_hint` directly? (Line 5) 

Thank you for answering!

**Comment by yuxineverforever** [comment]
Could anyone help with this question? @scarbain @williamyang1991 @lllyasviel @eltociear 

Thank you!

**Comment by Chen-Suyi** [comment]
I am trying to implement my own ControlNet, and I have the same question.

**Comment by zeezou-isee** [comment]
I agree with you @yuxineverforever, according to the model chart in the paper, I think before going through controlnet,` h += guided_hint` should be computed first.
so the code should be like bellow :
              ```
                h += guided_hint                
                h = module(h,emb,context)
```

**Comment by Twinkle-ce** [comment]
It seems that the first module in the network might be a convolutional layer rather than input_blocks. The input tensor h has 4 channels, while guided_hint has 320 channels, leading to a mismatch. My understanding is that h first passes through the initial module to adjust its channel dimension to 320 before being added to guided_hint.

If my interpretation is incorrect, I would appreciate any clarification. Looking forward to discussing this further.

Thanks! 😊
